### PR TITLE
Migrate to build machinery

### PR DESCRIPTION
**Pull request by tnozicka**
- other operators use build machinery from library-go
- you get ldflags for version, verify-deps and other benefits
- we keep the shared code updated in one repo (library-go)

/assign @adambkaplan 

replaces: https://github.com/openshift/release/pull/4308

### CODE REVIEW

**Comment by sttts** [comment]
/hold

Leaving finally decission to @adambkaplan

**Comment by tnozicka** [comment]
/hold
for @adambkaplan to bless it

File: manifests/02_config.crd.yaml-merge-patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# this file can be removed once we switch to v0.2 of crd generator
+# see: https://github.com/openshift/cluster-kube-scheduler-operator/pull/148
```
Comment (sttts) [inline manifests/02_config.crd.yaml-merge-patch]: @damemi stange comment. I don't see the pattern annotation in the api types. Maybe we should add it and get rid of these patches.

Comment (damemi) [reply manifests/02_config.crd.yaml-merge-patch]: https://github.com/openshift/api/blob/master/operator/v1/types.go#L27 ?

**Pull request merged**